File: xcloak-agent-desktop/agent/hunt_workbench_test.go
package agent

import (
	"encoding/json"
	"fmt"
	"testing"
)

// TestHuntTemplate_URLPatterns verifies correct URL construction for hunt
// workbench operations.
func TestHuntTemplate_URLPatterns(t *testing.T) {
	cases := []struct {
		got  string
		want string
	}{
		{fmt.Sprintf("/api/hunt/templates/%d", 2), "/api/hunt/templates/2"},
		{fmt.Sprintf("/api/hunt/runs/%d", 5), "/api/hunt/runs/5"},
		{fmt.Sprintf("/api/hunt/runs/%d/notes", 5), "/api/hunt/runs/5/notes"},
	}
	for _, c := range cases {
		if c.got != c.want {
			t.Errorf("URL = %q, want %q", c.got, c.want)
		}
	}
}

// TestHuntRun_PayloadFields verifies hunt run record has fields expected by
// the workbench UI.
func TestHuntRun_PayloadFields(t *testing.T) {
	raw := `{"id":1,"query_type":"process","query_text":"nmap","status":"completed","result_count":3,"started_at":"2025-01-01T00:00:00Z"}`
	var run map[string]any
	if err := json.Unmarshal([]byte(raw), &run); err != nil {
		t.Fatalf("unmarshal: %v", err)
	}
	for _, key := range []string{"id", "query_type", "query_text", "status", "result_count"} {
		if _, ok := run[key]; !ok {
			t.Errorf("hunt run payload missing field %q", key)
		}
	}
}

// TestUpdateHuntRunNotes_BodyShape verifies the PATCH notes body includes both
// notes and severity as the backend expects.
func TestUpdateHuntRunNotes_BodyShape(t *testing.T) {
	body := map[string]any{
		"notes":    "Found suspicious process on host-01",
		"severity": "high",
	}
	b, _ := json.Marshal(body)
	var decoded map[string]any
	json.Unmarshal(b, &decoded)
	for _, key := range []string{"notes", "severity"} {
		if _, ok := decoded[key]; !ok {
			t.Errorf("hunt run notes PATCH body missing field %q", key)
		}
	}
}

File: xcloak-agent-mobile/test/hunt_workbench_test.dart
// Tests for hunt workbench API shapes consumed by the mobile agent.
// The mobile agent read-only view shows hunt run summaries; notes and severity
// are set by analysts via the web UI and read back on mobile.
//
// Run with: flutter test

import 'package:flutter_test/flutter_test.dart';

void main() {
  group('Hunt Workbench (mobile API shape)', () {
    test('hunt run record includes required display fields', () {
      final run = {
        'id': 1,
        'query_type': 'process',
        'query_text': 'nmap',
        'status': 'completed',
        'result_count': 3,
        'severity': 'high',
        'notes': 'Found suspicious process',
        'started_at': '2025-01-01T00:00:00Z',
      };

      for (final key in ['id', 'query_type', 'status', 'result_count']) {
        expect(run.containsKey(key), isTrue,
            reason: 'hunt run must include "$key" for the workbench UI');
      }
    });

    test('hunt template record includes name and query fields', () {
      final tmpl = {
        'id': 1,
        'name': 'Nmap Scanner Detection',
        'query_type': 'process',
        'query_text': 'nmap',
        'description': 'Detect nmap port scans',
        'tags': ['recon', 'network'],
      };

      for (final key in ['id', 'name', 'query_type', 'query_text']) {
        expect(tmpl.containsKey(key), isTrue,
            reason: 'hunt template must include "$key"');
      }
    });

    test('update notes PATCH body includes notes and severity', () {
      final body = {
        'notes': 'Confirmed attacker recon activity',
        'severity': 'critical',
      };

      expect(body.containsKey('notes'), isTrue);
      expect(body.containsKey('severity'), isTrue);
      expect(body['notes'], isA<String>());
      expect(body['severity'], isA<String>());
    });
  });
}
